### PR TITLE
Integrate LLVM at llvm/llvm-project@5101e7e8dd017

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
Update BUILD files and submodule for
[5101e7e8dd017](https://github.com/llvm/llvm-project/commit/5101e7e8dd017)

### CODE REVIEW

File: llvm-bazel/llvm-project-overlay/llvm/BUILD [hunk 1]
```diff
@@ -2141,6 +2141,7 @@ gentbl(
             ":Core",
             ":IPO",
             ":MC",
+            ":Passes",
```
Comment (chandlerc) [inline llvm-bazel/llvm-project-overlay/llvm/BUILD]: I think this is an upstream layering violation -- can we wait to get it fixed?

I've sent a note to some folks about it already.

Comment (GMNGeoffrey) [reply llvm-bazel/llvm-project-overlay/llvm/BUILD]: Sure, although I think we could also merge this with a TODO and then delete it again when it's fixed? It seems good to keep these up to date.

Comment (chandlerc) [reply llvm-bazel/llvm-project-overlay/llvm/BUILD]: You're right.

**Review by chandlerc — COMMENTED** [review]
LGTM

**Review by chandlerc — APPROVED** [review]
Actually mark as LGTM this time...

File: llvm-bazel/llvm-project-overlay/llvm/BUILD (hunk 1 above)
Comment (GMNGeoffrey) [reply llvm-bazel/llvm-project-overlay/llvm/BUILD]: Good because I now get an email every 15 minutes that the integration branch is broken :-D

Comment (GMNGeoffrey) [reply llvm-bazel/llvm-project-overlay/llvm/BUILD]:
```suggestion
            ":Passes",  # TODO(chandlerc): Likely a layering violation.
```

Comment (GMNGeoffrey) [reply llvm-bazel/llvm-project-overlay/llvm/BUILD]: Lol GitHub.... The easiest way to edit in the UI is to create a suggested edit myself. If I go to edit the file it just pulls up the whole thing

**Comment by GMNGeoffrey** [comment]
RBE failure is because this PR predates the introduction of the RBE config

**Pull request merged**